apps/qubit/modules/search/actions/advancedAction.class.php

```diff
@@ -450,6 +462,65 @@ static function addToQueryBool(&$queryBool, $operator, $queryField)
       }
   }
 
+  static function getDateRangeQuery($start, $end)
+  {
+    if (empty($start) && empty($end))
+    {
+      return;
+    }
+
+    $query = new \Elastica\Query\Bool();
+    $range = array();
+
+    if (!empty($start))
```

Comment:
Hi Mike, it must be that way. If only one date is populated it will enter in only one if and add one query, but if both dates are populated it must enter in the three ifs and add the three queries. I did it like you're suggesting at the begining to sepparate the three cases more clearly, but in the end, this is the best way to avoid duplicated code/queries inside de ifs.